sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -0,0 +1,129 @@
+﻿# Azure Iot Hub Service API Design Doc
+This document outlines the APIs for the Azure Iot Hub Service SDK
+
+<details><summary><b>Type definition names</b></summary>
+    
+```
+Configuration - TwinConfiguration
+Module - ModuleIdentity
+Device - DeviceIdentity
+Twin - TwinData
+Interface - PnpInterface
+Property - PnpProperty
+Reported - PnpReported
+Desired - PnpDesired
+```
+</details>
+
+<details><summary><b>Constructors</b></summary>
+    
+```csharp
+
+```
+</details>
+
+<details><summary><b>Configurations</b></summary>
+APIs for managing configurations for devices and modules
+
+```csharp
+
+```
+</details>
+
+<details><summary><b>Statistics</b></summary>
+APIs for getting statistics about devices and modules, as well as service statistics
+
+```csharp
+
+```
+</details>
+
+<details><summary><b>Devices</b></summary>
+APIs for managing device identities, device twins, and querying devices
+
+```csharp
+
+```
+</details>
+
+<details><summary><b>Modules</b></summary>
+APIs for managing module identities, module twins, and querying modules
+
+```csharp
+
+```
+</details>
+
+<details><summary><b>Jobs</b></summary>
+APIs for using IotHub v2 jobs
+
+```csharp
+
+```
+</details>
+
+<details><summary><b>Messages</b></summary>
+Feedback messages, sending cloud to device messages (missing from current swagger), and purging cloud to device message queue
+
+```csharp
+public class CloudToDeviceMessages
+{
+    /// <summary>
+    /// Retrieve feedback notification for cloud to device messages.
```

Comment:
why call it feedback? it seems to be the "result" of the message sent from service to device